webtranslate/pages/string_edit.py

```diff
@@ -258,21 +277,37 @@ def find_string(pmd, lngname):
     if len(cur_strings) > len(sac.cache):
         cur_strings.difference_update(sac.cache.keys())
         sel = random.choice(list(cur_strings))
-        sac.add(sel)
-        return sel
     else:
         # Just try them all, and pick the best one.
         best_val = None
-        best_name = None
+        sel = None
         for sname in cur_strings:
             idx = sac.find(sname)
             if idx is None:
                 sac.add(sname)
                 return sname
             if best_val is None or best_val < idx:
-                best_val, best_name = idx, sname
-        sac.add(best_name)
-        return best_name
+                best_val, sel = idx, sname
+
+    sac.add(sel)
```

Comment:
I think the "_1" string should be added to the string-avoidance-cache instead.
That way the user either gets the first string in a block, or none from the block.